tests/test_frame.py

```diff
@@ -2526,3 +2526,51 @@ def test_loc_returns_series() -> None:
     df1 = pd.DataFrame({"x": [1, 2, 3, 4]}, index=[10, 20, 30, 40])
     df2 = df1.loc[10, :]
     check(assert_type(df2, Union[pd.Series, pd.DataFrame]), pd.Series)
+
+
+def test_to_dict_index() -> None:
+    df = pd.DataFrame({"a": [1, 2], "b": [9, 10]})
+    check(
+        assert_type(
+            df.to_dict(orient="records", index=True), List[Dict[Hashable, Any]]
+        ),
+        list,
+    )
+    check(assert_type(df.to_dict(orient="dict", index=True), dict[Hashable, Any]), dict)
+    check(
+        assert_type(df.to_dict(orient="series", index=True), dict[Hashable, Any]), dict
+    )
+    check(
+        assert_type(df.to_dict(orient="index", index=True), dict[Hashable, Any]), dict
+    )
+    check(
+        assert_type(df.to_dict(orient="index", index=True), dict[Hashable, Any]), dict
+    )
+    check(
+        assert_type(df.to_dict(orient="split", index=True), dict[Hashable, Any]), dict
+    )
+    check(
+        assert_type(df.to_dict(orient="tight", index=True), dict[Hashable, Any]), dict
+    )
+    check(
+        assert_type(df.to_dict(orient="tight", index=False), dict[Hashable, Any]), dict
+    )
+    check(
+        assert_type(df.to_dict(orient="split", index=False), dict[Hashable, Any]), dict
+    )
+    if TYPE_CHECKING_INVALID_USAGE:
+        check(assert_type(df.to_dict(orient="records", index=False), List[Dict[Hashable, Any]]), list)  # type: ignore[assert-type, call-overload] # pyright: ignore[reportGeneralTypeIssues]
+        check(assert_type(df.to_dict(orient="dict", index=False), dict[Hashable, Any]), dict)  # type: ignore[assert-type, call-overload] # pyright: ignore[reportGeneralTypeIssues]
+        check(assert_type(df.to_dict(orient="series", index=False), dict[Hashable, Any]), dict)  # type: ignore[assert-type, call-overload] # pyright: ignore[reportGeneralTypeIssues]
+        check(assert_type(df.to_dict(orient="index", index=False), dict[Hashable, Any]), dict)  # type: ignore[assert-type, call-overload] # pyright: ignore[reportGeneralTypeIssues]
+        check(assert_type(df.to_dict(orient="index", index=False), dict[Hashable, Any]), dict)  # type: ignore[assert-type, call-overload] # pyright: ignore[reportGeneralTypeIssues]
```

Comment:
this test is repeated.  2565 and 2566 are the same

Reply:
Done